### ISSUE TITLE
Typechecking doesn't allow for `raise` without arguments

### ISSUE THREAD

**Issue by wagenet** [castwide/solargraph#418]
```rb
begin
  raise StandardError, "error"
rescue
  puts "log something"
  raise # Reports: `Not enough arguments to Kernel#raise`
end
```

https://rubydoc.info/stdlib/core/Kernel:raise

**Comment by castwide** [comment]
I haven't been able to reproduce this problem. `Kernel#raise` is always seen as valid without arguments.

What versions of Solargraph and Ruby are you using?

**Comment by wagenet** [comment]
Ruby 2.7.2
Solargraph 0.40.3
Running `solargraph typecheck`

Here's some actual code that I'm seeing the issue with. The later two raise both cause the `Not enough arguments to Kernel#raise` notice.

```rb
  module Retries
    def with_retry(times: 1, error: StandardError)
      raise ArgumentError, "must retry at least one time" if times < 1

      begin
        yield
      rescue Exception => e
        raise unless Array(error).any? { |match_error| match_error === e }
        raise if times == 0

        times -= 1
        retry
      end
    end
  end
```

**Comment by castwide** [comment]
This appears to be a problem specifically with core documentation for Ruby 2.7+. The definition was previously `Kernel#raise(*)`, but now the definition is `Kernel#raise(*, _)`. The latter is incorrect.

As a temporary stopgap, I'll probably add a redefinition of `Kernel#raise` to the core fills.

**Comment by castwide** [comment]
Fixed in v0.43.0.
